index.py

```diff
@@ -186,6 +186,21 @@ def fetch(url: str):
     return text
 
 
+def _check_cloudflare(html: BeautifulSoup) -> None:
+    _possible_errors = ['denied', 'Cloudflare']
+    title = html.find('title')
+    has_cloudflare_issues = False
+    if title is not None:
+        for _possible_error in _possible_errors:
+            if _possible_error.lower() in title.text.lower():
+                has_cloudflare_issues = True
+                break
+
+    if has_cloudflare_issues:
+        pretty_print("!", "Failed because of Cloudflare protection!")
+        raise ValueError(title.text)

```

Comment:
This will for sure trigger the exception call at the end of the loop

Reply:
Yes, it will trigger the exception. But all program from is wrapped with `try... except` block, so that exception will be handled properly.
What do you think?